src/nash/mod.rs

```diff
@@ -706,73 +711,104 @@ impl Stream for NashStream {
-            return _stream.unwrap_err();
-        }
+    async fn create_stream_specific(
+        &self,
+        subscriptions: Subscriptions<Self::Subscription>,
+    ) -> Result<BoxStream<'static, Result<Self::Response>>> {
+        let mut streams = SelectAll::new();
 
-        Ok(())
-    }
+        for subscription in subscriptions.into_iter() {
+            let stream = Client::subscribe_protocol(&self.client, subscription.clone()).await;
+            let stream = stream.map_err(|e| Err(OpenLimitError::NashProtocolError(e)));
+
+            match stream {
+                Ok(s) => {
+                    streams.push(s);
+                }
+                Err(_) => {
+                    return stream.unwrap_err();
+                }
+            }
+        }
 
-    fn parse_message(&self, message: Self::Item) -> Result<OpenLimitsWebsocketMessage> {
-        match message {
+        let s = streams.map(|message| match message {
             Ok(msg) => match msg {
-                ResponseOrError::Response(resp) => Ok(resp.data.into()),
+                ResponseOrError::Response(resp) => Ok(SubscriptionResponseWrapper(resp.data)),
                 ResponseOrError::Error(resp) => {
                     let f = resp
                         .errors
                         .iter()
                         .map(|f| f.message.clone())
                         .collect::<Vec<String>>()
                         .join("\n");
-                    Err(OpenLimitError::NotParsableResponse(String::from(f)))
+                    Err(OpenLimitError::NotParsableResponse(f))
                 }
             },
             Err(_) => Err(OpenLimitError::SocketError()),
-        }
+        });
+
+        Ok(s.boxed())
     }
 }
 
 impl From<Subscription> for nash_protocol::protocol::subscriptions::SubscriptionRequest {
     fn from(sub: Subscription) -> Self {
         match sub {
-            Subscription::OrderBook(market, _depth) => Self::Orderbook(
+            Subscription::OrderBookUpdates(market) => Self::Orderbook(
                 nash_protocol::protocol::subscriptions::updated_orderbook::SubscribeOrderbook {
                     market,
                 },
             ),
-            Subscription::Trade(market) => Self::Trades(
+            Subscription::Trades(market) => Self::Trades(
                 nash_protocol::protocol::subscriptions::trades::SubscribeTrades { market },
             ),
             _ => panic!("Not supported Subscription"),
         }
     }
 }
 
-impl From<nash_protocol::protocol::subscriptions::SubscriptionResponse>
-    for OpenLimitsWebsocketMessage
-{
-    fn from(message: nash_protocol::protocol::subscriptions::SubscriptionResponse) -> Self {
-        match message {
-            nash_protocol::protocol::subscriptions::SubscriptionResponse::Orderbook(resp) => {
-                OpenLimitsWebsocketMessage::OrderBook(OrderBookResponse {
+#[derive(Debug)]
+pub struct SubscriptionResponseWrapper(SubscriptionResponse);
```

Comment:
It would be nice to make `nash_protocol::protocol::SubscriptionResponse` impl `Clone` to avoid this wrapper

Reply:
yes, can definitely do that!